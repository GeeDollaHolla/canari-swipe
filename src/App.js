import React, { Component } from 'react'
import Swiper from './Swiper'

const pageWidth = 300
const pageHeight = 200
export default class App extends Component {
	render() {
		const wrapperStyle = {
			height: pageHeight + 'px',
			width: pageWidth + 'px',
			overflow: 'hidden',
			display: 'inline-block',
			margin: 'auto',
			left: 0,
			right: 0,
		}

		const pageStyle = {
			height: pageHeight + 'px',
			width: pageWidth + 'px',
			backgroundColor: 'gray',
			display: 'inline-block',
			border: '3px solid white',
			color: 'white',
			textAlign: 'center',
			verticalAlign: 'middle',
			lineHeight: pageHeight + 'px',
			userSelect: 'none',
		}

		return (
			<div>
				<SingleSelection pageStyle={pageStyle} />
				<Carousel pageStyle={pageStyle} />
			</div>
		)
	}
}

const SingleSelection = ({ pageStyle }) => (
	<div>
		<h3>Single Selections</h3>

		<Swiper
			swipeAmount={pageWidth}
			visibleCount={1}
			horizontal={true}
			carousel={false}
			minimumSwipeSpeed={2000}
			wrapAround={true}
			desiredSelection={0}
			firstSelection={0}>
			{[...Array(5)].map((iter, page) => {
				return (
					<div key={page}>
						<div style={pageStyle}>Page {page}</div>
					</div>
				)
			})}
		</Swiper>
	</div>
)

const Carousel = ({ pageStyle }) => (
	<div>
		<h3>Carousel</h3>

		<Swiper
			swipeAmount={pageWidth}
			visibleCount={1}
			horizontal={true}
			carousel={true}
			minimumSwipeSpeed={2000}
			wrapAround={false}
			desiredSelection={0}
			firstSelection={0}>
			{[...Array(5)].map((iter, page) => {
				return (
					<div key={page}>
						<div style={pageStyle}>Page {page}</div>
					</div>
				)
			})}
		</Swiper>
	</div>
)
