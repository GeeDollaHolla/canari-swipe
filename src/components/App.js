import React, { Component } from 'react'
import SingleSelection from '../components/SingleSelection'
import Carousel from '../components/Carousel'
import Vertical from '../components/Vertical'
import Responsive from '../components/Responsive'

const pageWidth = 300
const pageHeight = 100
export default class App extends Component {
	render() {
		const pageStyle = {
			maxWidth: '800px',
			margin: 'auto',
			fontFamily: 'Calibri',
		}

		const selectionStyle = {
			height: pageHeight + 'px',
			width: pageWidth + 'px',
			backgroundColor: 'gray',
			display: 'inline-block',
			border: '3px solid white',
			color: 'white',
			textAlign: 'center',
			verticalAlign: 'middle',
			lineHeight: pageHeight + 'px',
			userSelect: 'none',
			fontFamily: 'calibri',
		}

		return (
			<div style={pageStyle}>
				<h1>canari-swipe examples</h1>
				<p>A few basic examples for canari-swipe</p>
				<a href="https://github.com/GeeDollaHolla/canari-swipe/blob/master/src/App.js">
					Source Code
				</a>
				<SingleSelection selectionStyle={selectionStyle} />
				<Carousel selectionStyle={selectionStyle} />
				<Vertical selectionStyle={selectionStyle} />
				<Responsive selectionStyle={selectionStyle} />
				<div style={{ display: 'inline-block', height: '200px' }} />
			</div>
		)
	}
}








