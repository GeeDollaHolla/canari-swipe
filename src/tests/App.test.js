import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import App from '../components/App'

test('renders correctly', () => {
	const wrapper = shallow(<App />)
	expect(toJson(wrapper)).toMatchSnapshot()
})
