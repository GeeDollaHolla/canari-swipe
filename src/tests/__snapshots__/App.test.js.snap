// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "fontFamily": "Calibri",
      "margin": "auto",
      "maxWidth": "800px",
    }
  }
>
  <h1>
    canari-swipe examples
  </h1>
  <p>
    A few basic examples for canari-swipe
  </p>
  <a
    href="https://github.com/GeeDollaHolla/canari-swipe/tree/master/src/components"
  >
    Source Code
  </a>
  <SingleSelection
    selectionStyle={
      Object {
        "backgroundColor": "gray",
        "border": "3px solid white",
        "color": "white",
        "display": "inline-block",
        "fontFamily": "calibri",
        "height": "100px",
        "lineHeight": "100px",
        "textAlign": "center",
        "userSelect": "none",
        "verticalAlign": "middle",
        "width": "300px",
      }
    }
  />
  <Carousel
    selectionStyle={
      Object {
        "backgroundColor": "gray",
        "border": "3px solid white",
        "color": "white",
        "display": "inline-block",
        "fontFamily": "calibri",
        "height": "100px",
        "lineHeight": "100px",
        "textAlign": "center",
        "userSelect": "none",
        "verticalAlign": "middle",
        "width": "300px",
      }
    }
  />
  <Vertical
    selectionStyle={
      Object {
        "backgroundColor": "gray",
        "border": "3px solid white",
        "color": "white",
        "display": "inline-block",
        "fontFamily": "calibri",
        "height": "100px",
        "lineHeight": "100px",
        "textAlign": "center",
        "userSelect": "none",
        "verticalAlign": "middle",
        "width": "300px",
      }
    }
  />
  <Responsive
    selectionStyle={
      Object {
        "backgroundColor": "gray",
        "border": "3px solid white",
        "color": "white",
        "display": "inline-block",
        "fontFamily": "calibri",
        "height": "100px",
        "lineHeight": "100px",
        "textAlign": "center",
        "userSelect": "none",
        "verticalAlign": "middle",
        "width": "300px",
      }
    }
  />
  <div
    style={
      Object {
        "display": "inline-block",
        "height": "200px",
      }
    }
  />
</div>
`;
